deployment/utils.py

```diff
@@ -0,0 +1,99 @@
+import os
+import typing
+from pathlib import Path
+from typing import List
+
+from ape import networks
+from ape.api import AccountAPI
+from ape.cli import get_user_selected_account
+from ape.contracts import ContractInstance
+
+from deployment.constants import (
+    CURRENT_NETWORK,
+    ETHERSCAN_API_KEY_ENVVAR,
+    LOCAL_BLOCKCHAIN_ENVIRONMENTS,
+    WEB3_INFURA_API_KEY_ENVVAR
+)
+from deployment.params import (
+    ApeDeploymentParameters,
+    ConstructorParameters
+)
+
+
+def check_registry_filepath(registry_filepath: Path) -> None:
+    """
+    Checks that the registry_filepath does not exist,
+    and that its parent directory does exist.
+    """
+    if registry_filepath.exists():
+        raise FileExistsError(f"Registry file already exists at {registry_filepath}")
+    if not registry_filepath.parent.exists():
+        raise FileNotFoundError(f"Parent directory of {registry_filepath} does not exist.")
+
+
+def check_etherscan_plugin() -> None:
+    """Checks that the ape-etherscan plugin is installed and that the ETHERSCAN_API_KEY is set."""
+    if CURRENT_NETWORK in LOCAL_BLOCKCHAIN_ENVIRONMENTS:
+        # unnecessary for local deployment
+        return
+
+    try:
+        import ape_etherscan  # noqa: F401
+    except ImportError:
+        raise ImportError("Please install the ape-etherscan plugin to use this script.")
+
+    api_key = os.environ.get(ETHERSCAN_API_KEY_ENVVAR)
```

Comment:
I'd suggest using ape-etherscan mapping for this, which is network-dependent. See https://github.com/nucypher/nucypher-contracts/blob/972ca1a6983a3d0fbc5e7966a0f333715f64f97d/scripts/deploy_flat_rate_fee_model.py#L40C23-L40C42